content/api/commands/origin.md

```diff
@@ -0,0 +1,467 @@
+})
+```
+
+### Navigating to multiple secondary origins in succession
+
+Callbacks may **not** themselves contain `cy.origin()` calls, so when visiting
+multiple origins, do so at the top level of the test.
+
+```js
+cy.origin('foo.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'foo.com')
+})
+
+cy.origin('bar.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'bar.com')
+})
+
+cy.origin('baz.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'baz.com')
+})
+```
+
+TODO Stabilization?
+
+### Waiting to return to primary origin
+
+Sometimes, a secondary origin returns to the primary origin as a result of user
+action (for example, by clicking a "Login" button on a syndicated login
+provider). In this situation, your test should wait for the navigation to
+complete before making further assertions, otherwise a cross-domain error will
+be thrown.
+
+```js
+cy.visit('/home')
+// This will take us to the secondary origin
+cy.contains('button', 'Go to someothersite').click()
+
+cy.origin('someothersite.com', () => {
+  // Click button that takes us back to primary origin
+  cy.contains('button', 'Go back').click()
+)
+// Wait until confirmation we are back at the primary origin before continuing
+cy.url().should('contain', '/home')
+// Do more things in primary domain...
+```
+
+### SSO login custom command
+
+A very common requirement is logging in to a site before running a test. If
+login itself is not the specific focus of the test, it's good to encapsulate
+this functionality in a `login`
+[custom command](/api/cypress-api/custom-commands) so you don't have to
+duplicate this login code in every test. Here's an idealized example of how to
+do this with `cy.origin()`.
+
+```js
+Cypress.Commands.add('login', (username, password) => {
+  // Remember to pass in dependencies via `args`
+  const args = { username, password }
+  cy.origin('auth-provider.com', { args }, ({ username, password }) => {
+    // Go to https://auth-provider.com/login
+    cy.visit('/login')
+    cy.contains('Username').find('input').type(username)
+    cy.contains('Password').find('input').type(password)
+    cy.get('button').contains('Login').click()
+  })
+  // Wait until confirmation we are back at the primary origin before continuing
+  cy.url().should('contain', '/home')
+})
+```
+
+However, having to go through an entire login flow before every test is not very
+performant. Up until now you could get around this by putting login code in the
+first test of your file, then performing subsequent tests reusing the same
+session. However, once the `experimentalSessionAndOrigin` flag is activated this
+is no longer possible, as all session state is now cleared between tests. So to
+avoid this overhead we have added the [`cy.session()`](/api/commands/session)
+command, which allows you to easily cache session information and reuse it
+across not just tests, but test files too. So now let's enhance our custom login
```

Comment:
session info is NOT available across test files. It resets across test files. https://github.com/cypress-io/cypress/issues/17710

```suggestion
across tests. So now let's enhance our custom login
```

Reply:
D'oh!